src/components/App.js

```diff
@@ -173,6 +169,38 @@ class App extends Component<{}, State> {
     setNotes(copy)
   }
 
+  removeFolder = (folder: Folder) => {
+    const archiveCopy = this.state.archive.slice()
+    const notesCopy = this.state.notes.slice()
+    const archiveTemp = []
+    this.state.notes.forEach(note => {
+      if (note.folder.name === folder) {
```

Comment:
I've seen this pattern in a couple of places--a `forEach` on `notes` looking for a set of folders with a particular name. I think the other place I've seen this pattern is in `changeFolderNames`.

Some thoughts:
1. Would it be helpful to encapsulate this in a function, perhaps using Array.prototype.filter?
2. Should the folder type also be checked? example problem case: what if a folder is called "archive" but is just a "normal" folder?
3. It looks to me like this is comparing the `note.folder.name` (of type `string`) with the `folder` (of type `Folder`)... should it be comparing with `folder.name`?